File: DiningPhilosophers.Sim/Model/StateDto.cs
using System;
using DiningPhilosophers.Core.Model;

namespace DiningPhilosophers.Sim.Model
{
    public class StateDto
    {
        public Guid TableId { get; set; }

        public TableType TableType { get; set; }

        public int PhilosopherId { get; set; }

        public bool IsDeadlockDetected { get; set; }

        public DateTime StartTime { get; set; }

        public DateTime? EndTime { get; set; }
        
        public TotalStats TotalStats { get; set; }
    }
}

File: DiningPhilosophers.WebClient/src/app/domain/state-dto.ts
import { TotalStats } from './total-stats';

// C#: DiningPhilosophers.Sim.Model.StateDto
export class StateDto {
    public tableId: string;

    public tableType: string;

    public philosopherId: number;

    public isDeadlockDetected: boolean;

    public startTime: string;

    public endTime?: string;

    public totalStats: TotalStats;
}
